templates/profile.php

```diff
@@ -14,14 +14,15 @@ function pmpromd_profile_preheader()
 		$main_post_id = $post->ID;
 
 		//Get the profile user
-		if(!empty($_REQUEST['pu']) && is_numeric($_REQUEST['pu']))
-			$pu = get_user_by('id', $_REQUEST['pu']);
-		elseif(!empty($_REQUEST['pu']))
-			$pu = get_user_by('slug', $_REQUEST['pu']);
-		elseif(!empty($current_user->ID))
+		$user_identifier = apply_filters( 'pmpromd_user_identifier', 'user_nicename' );

```

Comment:
This is used in multiple places, let's make a new function that has the hook in that. See line 20 for further direction since we can use another function to get the user and not have to call the identifier info more.